internal/metrics/reporter_test.go

```diff
@@ -0,0 +1,110 @@
+package metrics_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/lightstep/lightstep-tracer-common/golang/gogo/metricspb"
+	"github.com/lightstep/lightstep-tracer-go/internal/metrics"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Reporter", func() {
+	var reporter *metrics.Reporter
+	var ingestRequest metricspb.IngestRequest
+
+	JustBeforeEach(func() {
+		h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			body, _ := ioutil.ReadAll(r.Body)
+			err := proto.Unmarshal(body, &ingestRequest)
+			if !Expect(err).To(BeNil()) {
+				return
+			}
+		})
+		s := httptest.NewServer(h)
```

Comment:
Does this server not get closed? Looks like it leaks across tests?

Reply:
Good catch. Fixed